src/coreclr/vm/runtimehandles.cpp

```diff
@@ -1760,8 +1820,12 @@ FCIMPL1(INT32, RuntimeMethodHandle::GetSlot, MethodDesc *pMethod) {
 }
 FCIMPLEND
 
-FCIMPL3(Object *, SignatureNative::GetCustomModifiers, SignatureNative* pSignatureUNSAFE,
-    INT32 parameter, CLR_BOOL fRequired)
+FCIMPL5(Object *, SignatureNative::GetCustomModifiers,
+    SignatureNative* pSignatureUNSAFE,
+    INT32 rootSignatureParameterIndex,
```

Comment:
There are more problems with accounting for the various indexes. For example, this should print `CallConvCDecl`, but it prints nothing:
```
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;

var mt = typeof(My<delegate* <int, int, int>[]>).GetProperty("MyProp").GetModifiedPropertyType();
var f = mt.GetFunctionPointerParameterTypes()[1];
foreach (var cc in f.GetFunctionPointerCallingConventions())
    Console.WriteLine(cc.ToString());

public unsafe class My<T>
{
    public delegate* <T, delegate* unmanaged[Cdecl]<void>, void> MyProp { get; }
}
```

Reply:
I have pushed a commit to replace the nested indices in CoreCLR with a simple offset into signature that is incrementally updated. It fixes the bug above and makes the modified type tree initialized lazily as a bonus. Could you please take a look and add also add the above as a regression test?

If there is anything in my commit that you do not like, feel free to change it. (I have not paid too much attention to the style.)

Reply:
Sure thanks. This does lazy load on the tree and eager calculation of offset and caches the offset on the node. The prior did eager on tree (which could have been made lazy) and lazy calculation of offset (and did not cache the offset). The main benefit here as I see it is to avoid having the CLR keep an exact, common understanding of how the 3 indices work on the modified types. A secondary benefit is that subsequent access to the types\modifiers is more efficient due to the caching of the offsets.